pkg/apis/resourcemanager/v1alpha1/corporationtypeconfig_types.go

```diff
@@ -0,0 +1,97 @@
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CorporationTypeConfigSpec defines the desired state of CorporationTypeConfig
+// +k8s:protobuf=true
+type CorporationTypeConfigSpec struct {
+	// Available corporation types that can be selected for vendors
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	CorporationTypes []CorporationTypeDefinition `json:"corporationTypes"`
```

Comment:
I'd recommend moving to a **CorporationDefinition** type that has a spec and a status field. Each corporate definition would represent each corporation type that's available in the system (e.g. llc, inc, etc).